### PR TITLE
LUN-12: Refactor css of component

### PR DESCRIPTION
**Pull request by ansamshtaiwi1**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

File: src/components/Loading.tsx [hunk 1]
```diff
@@ -1,34 +1,32 @@
...
-    flex: 1,
-    alignItems: 'center'
-  },
-  indicator: {
-    height: '50%',
-    justifyContent: 'center',
-    alignItems: 'center'
-  }
-})
+const Root = styled.View`
```
Comment (SteffiStuffel) [inline src/components/Loading.tsx:6]: Something like LoadingContainer would be a better name.

File: src/components/MenuItem.tsx [hunk 2]
```diff
@@ -1,96 +1,77 @@
...
-  left: { flexDirection: 'row', alignItems: 'center' }
-})
+const Wrapper = styled.View`
+   padding-left: 16;
+   padding-right: 16;
+`;
+const ItemStyle = styled(Pressable)`
+    justify-content: space-between;
+    align-self: center;
+    padding-top: 17;
```
Comment (SteffiStuffel) [inline src/components/MenuItem.tsx:15]: Just merge paddings into one expression `padding: 17px 8px 17px 16px;`

File: src/components/MenuItem.tsx [hunk 3]
```diff
@@ -1,96 +1,77 @@
...
-  },
-  left: { flexDirection: 'row', alignItems: 'center' }
-})
+const Wrapper = styled.View`
+   padding-left: 16;
+   padding-right: 16;
+`;
+const ItemStyle = styled(Pressable)`
+    justify-content: space-between;
+    align-self: center;
```
Comment (SteffiStuffel) [inline src/components/MenuItem.tsx:14]: align-self can be removed i guess

File: src/components/MenuItem.tsx [hunk 4]
```diff
@@ -1,96 +1,77 @@
...
+    padding-top: 17;
+    padding-bottom: 17;
+    padding-right: 8;
+    padding-left: 16;
+    margin-bottom: 8;
+    flex-direction: row;
+    align-items: center;
+    width: 100%;
```
Comment (SteffiStuffel) [inline src/components/MenuItem.tsx:22]: can be removed i guess

File: src/components/MenuItem.tsx [hunk 5]
```diff
@@ -1,96 +1,77 @@
...
+    margin-left: ${(prop: IMenuItemProps) => prop.selected ? wp('5%'): 0 };
+    margin-right: ${(prop: IMenuItemProps) => prop.selected ? wp('5%'): 0 };
+    background-color: ${(prop: IMenuItemProps) => prop.selected? COLORS.lunesBlack : COLORS.white };
+    border-color: ${(prop: IMenuItemProps) => prop.selected? COLORS.white : COLORS.lunesBlackUltralight };
+    border-width:  ${(prop: IMenuItemProps) => !prop.selected ? 1: 0 };
+    border-style: ${(prop: IMenuItemProps) => !prop.selected ? 'solid' : 'solid' };
```
Comment (SteffiStuffel) [inline src/components/MenuItem.tsx:28]: ternery operator does not really make sense here.

File: src/components/MenuItem.tsx [hunk 6]
```diff
@@ -1,96 +1,77 @@
...
+    padding-right: 8;
+    padding-left: 16;
+    margin-bottom: 8;
+    flex-direction: row;
+    align-items: center;
+    width: 100%;
+    margin-left: ${(prop: IMenuItemProps) => prop.selected ? wp('5%'): 0 };
+    margin-right: ${(prop: IMenuItemProps) => prop.selected ? wp('5%'): 0 };
+    background-color: ${(prop: IMenuItemProps) => prop.selected? COLORS.lunesBlack : COLORS.white };
+    border-color: ${(prop: IMenuItemProps) => prop.selected? COLORS.white : COLORS.lunesBlackUltralight };
```
Comment (SteffiStuffel) [inline src/components/MenuItem.tsx:26]: ternery does not really make sense here, as you put border-width to 0, so the color should not matter.

File: src/components/MenuItem.tsx [hunk 7]
```diff
@@ -1,96 +1,77 @@
...
-  },
-  icon: {
-    justifyContent: 'center',
-    marginRight: 10,
-    width: wp('7%'),
-    height: wp('7%')
-  },
-  left: { flexDirection: 'row', alignItems: 'center' }
-})
+const Wrapper = styled.View`
```
Comment (SteffiStuffel) [inline src/components/MenuItem.tsx:8]: I think you can remove the Wrapper and put the margin into the ItemStyle.

File: src/components/MenuItem.tsx [hunk 8]
```diff
@@ -1,96 +1,77 @@
...
+    padding-top: 17;
+    padding-bottom: 17;
+    padding-right: 8;
+    padding-left: 16;
+    margin-bottom: 8;
+    flex-direction: row;
+    align-items: center;
+    width: 100%;
+    margin-left: ${(prop: IMenuItemProps) => prop.selected ? wp('5%'): 0 };
+    margin-right: ${(prop: IMenuItemProps) => prop.selected ? wp('5%'): 0 };
```
Comment (SteffiStuffel) [inline src/components/MenuItem.tsx:24]: I think you can just put `margin: 0px 16px 8px 16px;`and remove `margin-bottom: 8`

File: src/components/MenuItem.tsx [hunk 9]
```diff
@@ -1,96 +1,77 @@
...
+    align-items: center;
+`;
 
 export interface IMenuItemProps {
   selected: boolean
-  onPress: () => void
-  icon: string
-  title: string
-  children: ReactElement
+  onPress?: () => void
```
Comment (SteffiStuffel) [inline src/components/MenuItem.tsx:52]: Please undo this. If changes are needed to the prop type see the solution i suggested above (in buttons.tsx). It does not make sense to make the e.g. onPress optional, as a MenuItem definatly should have a onPress-Prop

File: src/components/MenuItem.tsx [hunk 10]
```diff
@@ -1,96 +1,77 @@
...
+  icon?: string
+  title?: string
+  children: ReactElement[] | string | undefined
```
Comment (SteffiStuffel) [inline src/components/MenuItem.tsx:55]: also undo this please.

File: src/components/MenuItem.tsx [hunk 11]
```diff
@@ -1,96 +1,77 @@
...
+    width: 100%;
+    margin-left: ${(prop: IMenuItemProps) => prop.selected ? wp('5%'): 0 };
+    margin-right: ${(prop: IMenuItemProps) => prop.selected ? wp('5%'): 0 };
+    background-color: ${(prop: IMenuItemProps) => prop.selected? COLORS.lunesBlack : COLORS.white };
+    border-color: ${(prop: IMenuItemProps) => prop.selected? COLORS.white : COLORS.lunesBlackUltralight };
+    border-width:  ${(prop: IMenuItemProps) => !prop.selected ? 1: 0 };
+    border-style: ${(prop: IMenuItemProps) => !prop.selected ? 'solid' : 'solid' };
+    border-radius: ${(prop: IMenuItemProps) => !prop.selected ? 2: 0 };
+`;
+const ItemTitleStyle = styled.Text`
```
Comment (SteffiStuffel) [inline src/components/MenuItem.tsx:31]: itemTitle would be enough as a name

File: src/components/MenuItem.tsx [hunk 12]
```diff
@@ -1,96 +1,77 @@
...
-    width: wp('7%'),
-    height: wp('7%')
-  },
-  left: { flexDirection: 'row', alignItems: 'center' }
-})
+const Wrapper = styled.View`
+   padding-left: 16;
+   padding-right: 16;
+`;
+const ItemStyle = styled(Pressable)`
```
Comment (SteffiStuffel) [inline src/components/MenuItem.tsx:12]: Just name it Item.

File: src/components/Modal.tsx [hunk 13]
```diff
@@ -10,61 +9,64 @@ import labels from '../constants/labels.json'
...
+const Message = styled.Text`
+    text-align: center;
+    font-size: ${wp('5%')};
+    color: ${COLORS.lunesGreyDark};
+    font-family: 'SourceSansPro-SemiBold';
+    width: ${wp('60%')};
+    margin-bottom: 31;
+    padding-top: 31;
+`;
+const LightLabel = styled.Text`
```
Comment (SteffiStuffel) [inline src/components/Modal.tsx:52]: here it ould make sense to have a generic class as Light and DarkLabel share all attributes but color. Or use props for setting the color.

File: src/components/Modal.tsx [hunk 14]
```diff
@@ -10,61 +9,64 @@ import labels from '../constants/labels.json'
...
+    flex: 1;
+    justify-content: center;
+    align-items: center;
+    margin-top: 0;
+    background-color: ${COLORS.lunesOverlay};
+`;
+const ModalStyle = styled.View`
+    background-color: ${COLORS.white};
+    align-items: center;
+    justify-content: center;
```
Comment (SteffiStuffel) [inline src/components/Modal.tsx:29]: justify-content does nothing

File: src/components/Modal.tsx [hunk 15]
```diff
@@ -10,61 +9,64 @@ import labels from '../constants/labels.json'
...
+const ModalContainer = styled.Modal`
+    flex: 1;
+    justify-content: center;
+    align-items: center;
+`;
+const Container = styled.View`
+    flex: 1;
+    justify-content: center;
+    align-items: center;
+    margin-top: 0;
```
Comment (SteffiStuffel) [inline src/components/Modal.tsx:23]: margin top does nothing

File: src/components/Modal.tsx [hunk 16]
```diff
@@ -10,61 +9,64 @@ import labels from '../constants/labels.json'
...
-    color: COLORS.lunesBlack,
-    fontSize: wp('4%'),
-    fontWeight: '600',
-    textAlign: 'center',
-    textTransform: 'uppercase',
-    fontFamily: 'SourceSansPro-SemiBold',
-    letterSpacing: 0.4
-  }
-})
+const ModalContainer = styled.Modal`
```
Comment (SteffiStuffel) [inline src/components/Modal.tsx:14]: Does noting just remove ModalContainer completely

File: src/components/Modal.tsx [hunk 17]
```diff
@@ -10,61 +9,64 @@ import labels from '../constants/labels.json'
...
+    flex: 1;
+    justify-content: center;
+    align-items: center;
+`;
+const Container = styled.View`
```
Comment (SteffiStuffel) [inline src/components/Modal.tsx:19]: Rename to Overlay please.

File: src/components/Modal.tsx [hunk 18]
```diff
@@ -10,61 +9,64 @@ import labels from '../constants/labels.json'
...
+    flex: 1;
+    justify-content: center;
+    align-items: center;
+    margin-top: 0;
+    background-color: ${COLORS.lunesOverlay};
+`;
+const ModalStyle = styled.View`
```
Comment (SteffiStuffel) [inline src/components/Modal.tsx:26]: Rename to ModalContainer please

**Review by SteffiStuffel — CHANGES_REQUESTED** [review]
Please rename PR so it starts with `LUN-12: Re....`

I made a lot of changeRequests here, feel free to contact me, why I suggested something if you don't see a benefit or suggest other solutions if you have something better in mind. We can also have a short call to discuss stuff or talk about this.

File: ios/Podfile.lock [hunk 19]
```diff
@@ -259,11 +259,11 @@ PODS:
     - React-Core
```
[the remark this answers is not in the document]
Comment (ansamshtaiwi1) [reply ios/Podfile.lock:259]: I'll commit all similar tasks together

[the remark this answers is not in the document]
Comment (ansamshtaiwi1) [reply ios/Podfile.lock:259]: Should I change anything here?

File: src/components/Button.tsx [hunk 20]
```diff
@@ -1,72 +1,48 @@
...
-    ...buttonBase,
-    backgroundColor: COLORS.lunesBlackUltralight
-  }
-})
-
+import styled from 'styled-components/native'
+
+const ButtonStyleTouchableOpacity = styled.TouchableOpacity`
+  flex-direction: row;
+  padding-top: 10;
```
[the remark this answers is not in the document]
Comment (ansamshtaiwi1) [reply src/components/Button.tsx:9]: done!

File: ios/Podfile.lock (hunk 19 above)
[the remark this answers is not in the document]
Comment (klinzo) [reply ios/Podfile.lock:259]: The changes to `RNSoundPlayer` and `TextToSpeech` are definitely fine, we forgot them in #127 . Cocoa Pods version should only be upgraded together with the version resolution in `ios/Gemfile` I think, I don't see a reason why we shouldn't do that though.

File: src/components/Button.tsx [hunk 21]
```diff
@@ -1,72 +1,48 @@
...
+  padding-bottom: 10;
+  padding-left: 20;
+  padding-right: 20;
+  width: ${wp('70%')};
+  align-items: center;
+  border-radius: ${hp('7%')};
+  justify-content: center;
+  height: ${hp('7%')};
+  margin-bottom: ${hp('2%')};
+  background-color: ${(prop: IButtonProps) => prop.disabled ? COLORS.lunesBlackUltralight :
```
[the remark this answers is not in the document]
Comment (ansamshtaiwi1) [reply src/components/Button.tsx:19]: is ### letter-spacing: 0.11 in em or px ?

[the remark this answers is not in the document]
Comment (ansamshtaiwi1) [reply src/components/Button.tsx:19]: px was added and ternery operator was replaced

File: src/components/Button.tsx [hunk 22]
```diff
@@ -1,72 +1,48 @@
...
-
+import styled from 'styled-components/native'
+
+const ButtonStyleTouchableOpacity = styled.TouchableOpacity`
+  flex-direction: row;
+  padding-top: 10;
+  padding-bottom: 10;
+  padding-left: 20;
+  padding-right: 20;
+  width: ${wp('70%')};
```
[the remark this answers is not in the document]
Comment (ansamshtaiwi1) [reply src/components/Button.tsx:13]: when I tried replace `wp('5%')`  to ` 5%` in` margin-left: ${(prop: IMenuItemProps) => prop.selected ? wp('5%'): 0 };` and similar, error appeared so I just left it as is.
do you have any suggestion?

[the remark this answers is not in the document]
Comment (SteffiStuffel) [reply src/components/Button.tsx:13]: you have to put it in single quotes `margin-right: ${(prop: IMenuItemProps) => prop.selected ? '5%': 0 };`

[the remark this answers is not in the document]
Comment (ansamshtaiwi1) [reply src/components/Button.tsx:13]: @Taggotty 
when I tried to remove wp/hp, the size of elements changed, should I change the percentage or keep wp/hp as it?

[the remark this answers is not in the document]
Comment (ansamshtaiwi1) [reply src/components/Button.tsx:13]: @Taggotty 
when I tried to remove wp/hp, the size of elements changed, should I change the percentage or keep wp/hp as it?

**Comment by Taggotty** [comment]
Feel free to commit and push your changes, so I can check how you progress or need any help.

**Review by SteffiStuffel — APPROVED** [review]

**Pull request merged**